xray/commands/curation/audit.go

```diff
@@ -0,0 +1,513 @@
+	getResp, respBody, _, err := nc.artiManager.Client().SendGet(packageUrl, true, &nc.httpClientDetails)
+	if err != nil {
+		if getResp == nil {
+			return nil, err
+		}
+		if getResp.StatusCode != http.StatusForbidden {
+			return nil, fmt.Errorf("Failed sending HEAD request to %s for package '%s:%s'. "+
+				"Status code: %v. Cause: %v", packageUrl, name, version, getResp.StatusCode, err)
+		}
+	}
+	if getResp.StatusCode == http.StatusForbidden {
+		respError := &ErrorsResp{}
+		if err := json.Unmarshal(respBody, respError); err != nil {
+			log.Error(fmt.Sprintf("failed to decode response for 'forbidden' response, err: %v", err))
+			return nil, nil
+		}
+		if len(respError.Errors) == 0 {
+			return nil, fmt.Errorf("No errors messages for package %s, version: %s for download url: %s ", name, version, packageUrl)
+		}
+		// if the error message contains the curation string key, then we can be sure it got blocked by curation service.
+		if strings.Contains(strings.ToLower(respError.Errors[0].Message), BlockMessageKey) {
+			blockingReason := BlockingReasonPolicy
+			if strings.Contains(strings.ToLower(respError.Errors[0].Message), NotBeingFoundKey) {
+				blockingReason = BlockingReasonNotFound
+			}
+			policies := extractPoliciesFromMsg(respError)
+			return &PackageStatus{
+				PackageName:       name,
+				PackageVersion:    version,
+				BlockedPackageUrl: packageUrl,
+				Action:            blocked,
+				Policy:            policies,
+				BlockingReason:    blockingReason,
+				PkgType:           string(nc.tech),
+			}, nil
+		}
+	}
+	return nil, nil
+}
+
+// Return policies and conditions names from forbidden error message. The error message structure happens when package was blocked by curation.
+// Message structure: Package %s:%s download was blocked by JFrog Packages Curation service due to the following policies violated {%s, %s},{%s, %s}.
+func extractPoliciesFromMsg(respError *ErrorsResp) []Policy {
+	var policies []Policy
+	msg := respError.Errors[0].Message
+	start := strings.Index(msg, "{")
+	end := strings.Index(msg, "}")
+	for end != -1 {
+		exp := msg[start:end]
+		exp = strings.TrimPrefix(exp, "{")
+		polCond := strings.Split(exp, ",")
+		if len(polCond) == 2 {
+			pol := polCond[0]
+			cond := polCond[1]
+			policies = append(policies, Policy{Policy: strings.TrimSpace(pol), Condition: strings.TrimSpace(cond)})
+		}
+		if len(msg) <= end+1 {
+			break
+		}
+		msg = msg[end+1:]
+		start = strings.Index(msg, "{")
+		end = strings.Index(msg, "}")
+	}
+	return policies
+}
+
+func getUrlNameAndVersionByTech(tech coreutils.Technology, nodeId, artiUrl, repo string) (downloadUrl string, name string, version string) {
+	switch tech {
+	case coreutils.Npm:
+		return getNameScopeAndVersion(nodeId, artiUrl, repo, coreutils.Npm.ToString())
+	}
+	return
+}
+
+// Graph holds for each node the component id(xray representation)
+// we extract from it packge name, version and contruct artifactory download url.
```

Comment:
Let's rephrase description to make it easier to read and correct in terms of grammar.